### DISCUSSION TITLE
How can I use matplotlib?

### DISCUSSION THREAD

**Discussion by TBR-DQ** [SUSTech-CS305-Fall22/CS305-Project-Skeleton#76] [Q&A] [answered]
I've already passed basic transfer test on vmware ubuntu 18.04 and now I'm trying to use matplotlib to draw a picture of windowsize. So I simply added one sentence
import matplotlib.pyplot as plt
and tries to run the test.
And I failed.

It is fine to run the test in command by using

python3 src/peer.py -p test/tmp2/nodes2.map -c test/tmp2/data1.fragment -m 1 -i 1 -t 60
python3 src/peer.py -p test/tmp2/nodes2.map -c test/tmp2/data2.fragment -m 1 -i 2 -t 60
DOWNLOAD test/tmp2/download_target.chunkhash test/tmp2/download_result.fragment

and by this way I successfully got my picture.

And it is also fine to run and pass the tests in 
basic_transfer_test.py
WITHOUT importing matplotlib and using matplotlib functions.

Although I got my picture, it seems that I won't pass the test if I add matplotlib functions into my code.
But without matplotlib functions I won't be able to show windowsize picture in peer.py.

So is there any way to fix this problem?
or I just does not need to add the picture-drawing part into my final code? Possibly submitting a picture result is fine?

**Comment by TANSixu** [comment, marked as answer]
Hi, thanks for the post!

We have noticed a significant delay of importing matplotlib in peer.py file, which could result in our grader's failure to start peer processes in time. You can refer to this post #60 to see if the solution there is helpful. 

It is fine to submit your code without plotting, as your plot will only be checked on the presentation.

If this issue still cannot be resolved, please follow up with the error message. [update: as well as snippets of peer1.log, peer2.log]
